src/librustdoc/passes/lint/redundant_explicit_links.rs

```diff
@@ -98,13 +98,8 @@ fn check_redundant_explicit_link<'md>(
 
                 let explicit_link = dest.to_string();
                 let display_link = link_data.resolvable_link.clone()?;
-                let explicit_len = explicit_link.len();
```

Comment:
Same logic in [collect_infra_links.rs](https://github.com/rust-lang/rust/blob/fe5f5912579d02c4e6088c89574493dfa988531a/src/librustdoc/passes/collect_intra_doc_links.rs#L1435) should be replaced with new logic.